gobblin-audit/src/main/java/gobblin/audit/auditor/ValueAuditGenerator.java

```diff
@@ -0,0 +1,99 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.audit.auditor;
+
+import gobblin.audit.policy.column.ColumnProjectionPolicy;
+import gobblin.audit.policy.row.RowSelectionPolicy;
+import gobblin.audit.sink.AuditSink;
+import gobblin.audit.sink.FsAuditSink;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Properties;
+
+import org.apache.avro.generic.GenericRecord;
+import org.apache.commons.lang3.reflect.ConstructorUtils;
+import org.apache.hadoop.fs.FileSystem;
+
+import com.google.common.base.Preconditions;
+
+
+/**
+ * The class that implements value based auditing. The class captures the values of certain
+ * columns from the rows in the dataset using the {@link ColumnProjectionPolicy}.
+ * This is done for every row or for a sample of the rows as defined by the {@link RowSelectionPolicy}
+ *
+ * The class initializes with two arguments
+ * <ul>
+ * <li> <code>jobProps</code>
+ *  <ul>
+ *    <li> <code>jobProps</code> must have the key {@value #COLUMN_PROJECTION_POLICY_CLASS_NAME_KEY} set to initialize the {@link ColumnProjectionPolicy}
+ *    <li> <code>jobProps</code> must have the key {@value #ROW_SELECTION_POLICY_CLASS_NAME_KEY} set to initialize the {@link RowSelectionPolicy}
+ *    <li> <code>jobProps</code> may have the key {@value #AUDIT_SINK_CLASS_NAME_KEY} set to initialize the {@link AuditSink}
+ *        or a default {@link FsAuditSink} for local {@link FileSystem} is used
+ * </ul>
+ * <li> <code>runtimeAuditMetadata</code> is used to pass the table specific runtime information like tablename, databaseName, snapshotName etc.
+ *      See {@link ValueAuditRuntimeMetadata}
+ * </ul>
+ *
+ * {@link ValueAuditGenerator#audit(GenericRecord)} is the method that audits an inputRecord.
+ */
+public class ValueAuditGenerator implements Closeable {
+
+  private static final String COLUMN_PROJECTION_POLICY_CLASS_NAME_KEY = "gobblin.audit.column.projection.class.name";
+  private static final String ROW_SELECTION_POLICY_CLASS_NAME_KEY = "gobblin.audit.row.selection.class.name";
+  private static final String AUDIT_SINK_CLASS_NAME_KEY = "gobblin.audit.sink.class.name";
+
+  private static final String DEFAULT_AUDIT_SINK_CLASS = "gobblin.audit.sink.FsAuditSink";
+
+  private final ColumnProjectionPolicy columnSelectionPolicy;
+  private final RowSelectionPolicy rowSelectionPolicy;
+  private final AuditSink auditSink;
+
+  public ValueAuditGenerator(Properties JobProps, ValueAuditRuntimeMetadata runtimeAuditMetadata) throws IOException {
```

Comment:
Please have the regular constructor just get the injected dependencies (the column and row selection policies and the sink). You can have a factory method that constructs an object using Properties  or even better, a typesafe config object.